packages/v3/src/use/components.js

```diff
@@ -0,0 +1,27 @@
+import { reactivePick } from '@vueuse/core';
+import { computed } from 'vue';
+
+/**
+ * Picks subcomponent props from super component props based on a subcomponent map
+ * @param props Reactive super component props
+ * @param subcomponents Map of kind name to component
+ * @param kindKey key under which to find the current kind name in props
+ */
+export const useSubcomponentProps = (props, subcomponents, kindKey = 'kind') =>
+  computed(() =>
+    reactivePick(props, Object.keys(subcomponents[props[kindKey]].props))
+  );
+
+/**
+ * Reactive subcomponent and sub props
+ * @param props Reactive super component props
+ * @param subcomponents Map of kind name to component
+ * @param kindKey key under which to find the current kind name in props
+ */
+export const useSubcomponent = (props, subcomponents, kindKey = 'kind') => ({
```

Comment:
I am a bit conflicted here. In the Vue2 version, we unused props would be passed through as $attrs and by default bound to the component. Here I believe we are discarding the props.

Reply:
We are indeed discarding the additional props, since adding them would lead to rendered HTML like this:

```html
<button type="button" class="bx--copy bx--snippet bx--snippet--inline" aria-live="polite" aria-label="" aria-describedby="8b2dc7ff-0934-498f-b0f4-c92679fbf0b2" kind="inline" wraptext="false" lesstext="Show less" moretext="Show more" maxcollapsednumberofrows="15" maxexpandednumberofrows="0" mincollapsednumberofrows="3" minexpandednumberofrows="16"><code id="8b2dc7ff-0934-498f-b0f4-c92679fbf0b2">node -v</code> Copied! <span aria-hidden="true" class="bx--assistive-text bx--copy-btn__feedback">Copied!</span></button>
```

Any attributes that are not part of the parent component props will automatically be applied to the resulting outer element. E.g. if you do `<cv-code-snippet kind="inline" asdf>`, the rendered `<button type="button"...>` would have the `asdf` attribute.

The only thing that can't be passed are attributes that are defined on the parent, but not on the child - but I think it's okay if we lose those.

Reply:
Understood.